browser/debounce/debounce_service_factory.cc

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/debounce/debounce_service_factory.h"
+
+#include <memory>
+#include <string>
+
+#include "base/memory/singleton.h"
+#include "brave/browser/brave_browser_process.h"
+#include "brave/components/debounce/browser/debounce_service.h"
+#include "brave/components/debounce/browser/debounce_service_impl.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/keyed_service/core/keyed_service.h"
+
+namespace debounce {
+
+// static
+DebounceServiceFactory* DebounceServiceFactory::GetInstance() {
+  return base::Singleton<DebounceServiceFactory>::get();
+}
+
+DebounceService* DebounceServiceFactory::GetForBrowserContext(
+    content::BrowserContext* context) {
+  return static_cast<DebounceService*>(
+      GetInstance()->GetServiceForBrowserContext(context, true));
+}
+
+DebounceServiceFactory::DebounceServiceFactory()
+    : BrowserContextKeyedServiceFactory(
+          "DebounceService",
+          BrowserContextDependencyManager::GetInstance()) {}
+
+DebounceServiceFactory::~DebounceServiceFactory() = default;
+
+KeyedService* DebounceServiceFactory::BuildServiceInstanceFor(
```

Comment:
it appears to me that if `GetBrowserContextToUse` is not overriden, this service won't be created for Incognito, is that intended?

Reply:
added it, returning `chrome::GetBrowserContextRedirectedInIncognito(context)` since there's nothing incognito- or non-incognito-specific about the service or its data.